microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/config/java/MicroProfileConfigHoverParticipant.java

```diff
@@ -101,57 +107,114 @@ public Hover collectHover(JavaHoverContext context, IProgressMonitor monitor) th
 			return null;
 		}
 
-		String propertyValue = JDTMicroProfileProjectManager.getInstance().getJDTMicroProfileProject(javaProject)
-				.getProperty(propertyKey, null);
-		if (propertyValue == null) {
-			propertyValue = getAnnotationMemberValue(annotation, CONFIG_PROPERTY_ANNOTATION_DEFAULT_VALUE);
-			if (propertyValue != null && propertyValue.length() == 0) {
-				propertyValue = null;
+		JDTMicroProfileProject mpProject = JDTMicroProfileProjectManager.getInstance()
+				.getJDTMicroProfileProject(javaProject);
+		List<MicroProfileConfigPropertyInformation> propertyInformation = getConfigPropertyInformation(propertyKey,
+				annotation, mpProject);
+		return new Hover(getDocumentation(propertyInformation, context.getDocumentFormat(),
+				context.isSurroundEqualsWithSpaces()), propertyKeyRange);
+	}
+
+	/**
+	 * Returns all the config property information for the given property key.
+	 *
+	 * Includes the information for all the different profiles.
+	 *
+	 * @param propertyKey              the property key without the profile
+	 * @param configPropertyAnnotation the annotation that defines the config property
+	 * @param project                  the project
+	 * @return the config property information for the given property key
+	 * @throws JavaModelException if an error occurs when accessing the config property annotation
+	 */
+	public static List<MicroProfileConfigPropertyInformation> getConfigPropertyInformation(String propertyKey,
+			IAnnotation configPropertyAnnotation, JDTMicroProfileProject project) throws JavaModelException {
+
+		List<MicroProfileConfigPropertyInformation> allInfo = project.collectPropertyInformation();

```

Comment:
I would prefer having this code:

```java
List<MicroProfileConfigPropertyInformation> neededInfo= project.getPropertyInformations(propertyKey);
```

to avoid creating a MicroProfileConfigPropertyInformation instances for all properties.